### PR TITLE
Update README on GitHub

### LINKED ISSUE #2939: Update README
Suggestion from @jougs to change the text in the README, remove a lot of text that's there and provide links to documentation / publications, and add text about setting up development environment (pre-commit hooks) etc.

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR updates the README since much of the information is redundant and we can point to documentation for more details. Some parts like citation and contribution are also available in the docs but it seems appropriate to keep some of this info here.

Fixes #2939

You can see how this looks here: https://github.com/jessica-mitchell/nest-simulator/tree/update-readme?tab=readme-ov-file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,109 +23,76 @@
...
+you can find [ways to get in touch here](https://nest-simulator.readthedocs.io/en/stable/community.html).
 
 
+## Contribute
+
+NEST is built on an active community and we welcome contributions to our code and documentation.
+
+You can create a [GitHub issue](https://github.com/nest/nest-simulator/issues/new/choose), for bug reports, feature requests, documentation improvements, or other issues.
+
+For working with NEST code and documentation, [see our documentation](https://nest-simulator.readthedocs.io/en/stable/developer_space/index.html#contribute-to-nest)
```
Comment (terhorstd) [inline README.md:87]: Also here we could add more links, especially to highlight the non-documentation channels, too.
```suggestion
NEST is built on an active community and we welcome everyone on [our mailinglist](https://nest-simulator.org/mailinglist) with comments and questions.

For documentation improvements, feature requests, bug reports or other issues you can create a [GitHub issue](https://github.com/nest/nest-simulator/issues/new/choose) which will be discussed in our fortnightly [meetings](https://nest-simulator.org/videoconference).

Active contributions to our code and documentation are highly regarded and you can find important steps for working with NEST code and documentation in our [contribute-to-nest documentation](https://nest-simulator.readthedocs.io/en/stable/developer_space/index.html#contribute-to-nest)
```

**Review by terhorstd — CHANGES_REQUESTED** [review]
Some smaller changes that could be considered. Not a strong requirement though. If there are reasons against this, of course feel free to reject them, or improve the language constructs where necessary.

File: README.md [hunk 2]
```diff
@@ -23,109 +23,76 @@
...
+## Key features of NEST
 
-A NEST simulation tries to follow the logic of an electrophysiological
-experiment that takes place inside a computer with the difference, that the
-neural system to be investigated must be defined by the experimenter.
+* NEST provides a Python interface or a stand-alone application
+* NEST provides a large collection of neurons and synapse models
+* NEST provides numerous example network scripts along with tutorials and guides to help you
+  develop your simulation
+* NEST has a large community of experienced developers and users; NEST was first released in 1994 under the name SYNOD, and has been extended and improved ever since
```
[the remark this answers is not in the document]
Comment (jessica-mitchell) [reply README.md:52]: I don't think so - but I have to double check 
The `/documentation` already goes to stable

**Review by terhorstd — APPROVED** [review]
Thanks for the work and the enduring push on this.
The text looks much better!

**Pull request merged**